### PR TITLE
Metrics UI changes/fixes

### PR DESCRIPTION
**Pull request by swati-sh**
### Requirements

* Metrics UI changes/fixes

### Description of the Change

Changing data structure from object to array to show the metrics data. Changes in filter functionality

### CODE REVIEW

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts [hunk 1]
```diff
@@ -48,11 +48,14 @@ export class ServiceMetricsComponent implements OnInit, AfterViewInit {
       column: 'View By:',
       label: 'PERIOD',
       type: 'select',
-      options: ['1 Minute', '1 Hour', '1 Day'],
-      values: [moment(0).add(1, 'minute').valueOf() / 1000,
-        moment(0).add(1, 'hour').valueOf() / 1000,
-        moment(0).add(1, 'day').valueOf() / 1000],
-      selected: '1 Minute'
+      options: ['15 Minutes', '1 Hour', '6 Hours', '1 Day', '7 Days', '30 Days'],
+      values: [moment(0).add(15, 'minute').valueOf() / 1000,
```
Comment (pspani) [inline core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: please check what values are supported for apigee vs API Gateway.

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts [hunk 2]
```diff
@@ -92,24 +94,10 @@ export class ServiceMetricsComponent implements OnInit, AfterViewInit {
...
 
-    })
   }
 
   ngOnInit() {
     this.serviceType = this.service.type || this.service.serviceType;
-    this.setPeriodFilters();
-  }
-
-  setPeriodFilters() {
```
Comment (supritat) [inline core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: @swati-sh Include this. It is used for GCP.

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts [hunk 3]
```diff
@@ -120,17 +108,19 @@ export class ServiceMetricsComponent implements OnInit, AfterViewInit {
     return this.http.get('/jazz/environments', {
       domain: this.service.domain,
       service: this.service.name
-    }, this.service.id).toPromise()
```
Comment (supritat) [inline core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: Include this.

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts [hunk 4]
```diff
@@ -262,34 +157,32 @@ export class ServiceMetricsComponent implements OnInit, AfterViewInit {
         statistics: this.filters.getFieldValueOfLabel('AGGREGATION')
       }
     };
-    return this.http.post(request.url, request.body, this.service.id)
```
Comment (supritat) [inline core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: Include this.

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts [hunk 5]
```diff
@@ -262,34 +157,32 @@ export class ServiceMetricsComponent implements OnInit, AfterViewInit {
...
+    this.allData = data;
+    this.queryDataRaw.assets = this.filterAssetType(this.allData[0]);
+    this.setAssetsFilter();
+    this.setAsset();
   }
-
   filterAssetType(data) {
     return data.assets.filter((asset) => {
       if (this.serviceType === 'api') {
-        if (this.service.deployment_targets === "gcp_apigee") {
```
Comment (supritat) [inline core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: Include this if-else conditions. (Looks like you need to take pull from upstream)

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts (hunk 2 above)
Comment (swati-sh) [reply core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: Done

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts (hunk 3 above)
Comment (swati-sh) [reply core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: Included

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts (hunk 4 above)
Comment (swati-sh) [reply core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: Done

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts (hunk 5 above)
Comment (swati-sh) [reply core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: Included this

File: core/jazz_ui/src/environments/environment.oss.ts [hunk 6]
```diff
@@ -1,33 +1,33 @@
 export const environment = {
   production: true,
-  INSTALLER_VARS: {INSTALLER_VARS},
+  INSTALLER_VARS: { "environment_tabs": { "overview": true, "code_quality": "false", "deployments": true, "assets": true, "logs": true }, "CREATE_SERVICE": { "DEPLOYMENT_TARGETS": { "API": { "options": [{ "label": "AWS API Gateway", "value": "aws_apigateway" }, { "label": "GCP APIGEE", "value": "gcp_apigee" }], "active": true }, "FUNCTION": { "options": [{ "label": "AWS Lambda", "value": "aws_lambda" }], "active": false }, "WEBSITE": { "options": [{ "label": "AWS S3", "value": "aws_s3" }, { "label": "AWS Cloudfront", "value": "aws_cloudfront" }], "active": false } } }, "feature": { "multi_env": true, "apigee": "false" }, "service_tabs": { "overview": true, "cost": false, "access_control": true, "metrics": true, "logs": true } },
   configFile: 'config/config.oss.json',
-  baseurl: "https://{API_GATEWAY_KEY_PROD}.execute-api.{inst_region}.amazonaws.com/prod",
-  api_doc_name: "https://{api_doc_name}.s3.amazonaws.com",
+  baseurl: "https://dbn9ruqpn3.execute-api.us-east-1.amazonaws.com/prod",
```
Comment (pspani) [inline core/jazz_ui/src/environments/environment.oss.ts]: remove hard coded values

File: core/jazz_ui/src/environments/environment.oss.ts [hunk 7]
```diff
@@ -1,33 +1,33 @@
...
+  api_doc_name: "https://multi0312-jazz-s3-api-doc-20190312053238719000000007.s3.amazonaws.com",
```
Comment (pspani) [inline core/jazz_ui/src/environments/environment.oss.ts]: remove hardcoded values, and in other places

Comment (swati-sh) [reply core/jazz_ui/src/environments/environment.oss.ts]: Removed

File: core/jazz_ui/src/app/secondary-components/filter-modal/filter-modal.component.ts [hunk 8]
```diff
@@ -134,27 +134,29 @@ export class FilterModalComponent implements OnInit {
...
     }
 
     let field = {
       column: column,
       label: label,
       options: options,
+      type: type,
       values: values || options,
-      selected: selected
+      selected: selected,
```
Comment (pspani) [inline core/jazz_ui/src/app/secondary-components/filter-modal/filter-modal.component.ts]: extra comma

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts [hunk 9]
```diff
@@ -366,16 +367,10 @@ export class ServiceMetricsComponent implements OnInit, AfterViewInit {
...
       })
       .map((dataPoint) => {
-
-        let obj = {
+        return {
           x: moment(dataPoint.Timestamp).valueOf(),
           y: parseInt(dataPoint[valueProperty])
         };
-
-        if(!obj['y']){
```
Comment (pspani) [inline core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: why was this removed?

Comment (swati-sh) [reply core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: These are not my changes.

File: core/jazz_ui/src/environments/environment.oss.ts (hunk 6 above)
Comment (swati-sh) [reply core/jazz_ui/src/environments/environment.oss.ts]: removed

File: core/jazz_ui/src/app/secondary-components/filter-modal/filter-modal.component.ts (hunk 8 above)
Comment (swati-sh) [reply core/jazz_ui/src/app/secondary-components/filter-modal/filter-modal.component.ts]: Removed

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts (hunk 9 above)
Comment (swati-sh) [reply core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: done the changes

File: core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts (hunk 1 above)
Comment (swati-sh) [reply core/jazz_ui/src/app/pages/service-metrics/service-metrics.component.ts]: checked
